src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs

```diff
@@ -0,0 +1,301 @@
+                    Assert.Equal("IntAdd", top_frame["functionName"].Value<string>());
+                    Assert.Contains("debugger-test.cs", top_frame["url"].Value<string>());
+
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 8, 4, scripts, top_frame["functionLocation"]);
+
+                    //now check the scope
+                    var scope = top_frame["scopeChain"][0];
+                    Assert.Equal("local", scope["type"]);
+                    Assert.Equal("IntAdd", scope["name"]);
+
+                    Assert.Equal("object", scope["object"]["type"]);
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 8, 4, scripts, scope["startLocation"]);
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 14, 4, scripts, scope["endLocation"]);
+                    return Task.CompletedTask;
+                }
+            );
+        }
+
+        [Fact]
+        public async Task ConditionalBreakpointTrue()
+        {
+            var bp1_res = await SetBreakpoint("dotnet://debugger-test.dll/debugger-test.cs", 12, 8, condition:"c == 30");
+
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_add(); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs", 12, 8, "IntAdd");
+        }
+
+        [Fact]
+        public async Task ConditionalBreakpointFalse()
+        {
+            var bp1_res = await SetBreakpoint("dotnet://debugger-test.dll/debugger-test.cs", 12, 8, condition:"c == 40");
+            var bp2_res = await SetBreakpoint("dotnet://debugger-test.dll/debugger-test.cs", 13, 8);
+
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_add(); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs", 13, 8, "IntAdd");
+        }
+
+        [Fact]
+        public async Task ConditionalBreakpointInvalidCondition() //testing with js debugger an invalid expression the debugger does not stop
+        {
+            var bp1_res = await SetBreakpoint("dotnet://debugger-test.dll/debugger-test.cs", 12, 8, condition:"g == 40");
+            var bp2_res = await SetBreakpoint("dotnet://debugger-test.dll/debugger-test.cs", 13, 8);
+
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_add(); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs", 13, 8, "IntAdd");
+        }
+
+        [Fact]
+        public async Task ConditionalBreakpointConst1()
+        {
+            var bp1_res = await SetBreakpoint("dotnet://debugger-test.dll/debugger-test.cs", 12, 8, condition:"true");
+
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_add(); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs", 12, 8, "IntAdd");
+        }
+
+        [Fact]
+        public async Task ConditionalBreakpointConst2()
```

Comment:
All the tests which are the same, except for the condition could use `Theory`+`InlineData`.

Reply:
Done.